src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts

```diff
@@ -801,7 +801,8 @@ export class WorkspaceRecommendedExtensionsView extends ExtensionsListView {
 	}
 
 	async show(query: string): Promise<IPagedModel<IExtension>> {
-		let model = await ((query && query.trim() !== '@recommended') ? this.showEmptyModel() : super.show(this.recommendedExtensionsQuery));
+		let shouldShowWorkspaceRecommended = query && (query.trim() === '@recommended' || query.trim() === '@recommended:workspace');
```

Comment:
When `query` is an empty string, this will end up showing the empty model, where as all the other views do not do so. So we are not being consistent